services/QuillLMS/engines/evidence/app/controllers/evidence/research/gen_ai/datasets_controller.rb

```diff
@@ -4,6 +4,8 @@ module Evidence
   module Research
     module GenAI
       class DatasetsController < ApplicationController
+        before_action :set_dataset, only: [:update]
```

Comment:
This is stylistic, so feel free to take or leave.  In the `dataset_relevant_texts_controller` you've defined one-line methods for `dataset` and `dataset_relevant_text` that set the values on first call, while in this controller you're using the `before_action` pattern to set the value on specific actions.  I think it might be worth picking one approach and sticking with it.  We use `before_action` in most of our older controllers, but have moved toward one-liners in a lot of our more recent Service-focused code, so I don't have a preference over one or the other.  It's not even wrong to use two different approaches if you want, either, but I figured I'd flag it in case the consistency matters to us.